k4MarlinWrapper/k4MarlinWrapper/util/EDM4hep2LcioUtils.h

```diff
@@ -0,0 +1,71 @@
+#ifndef EDM4HEP2LCIOUTILS_H
+#define EDM4HEP2LCIOUTILS_H
+
+namespace k4MW::util {
+
+// Reorder parameters based on dependencies between collections
+// This avoids (partly) the need for FillMissingCollections
+template <typename T>
+void optimizeOrderParams(
+  T& params)
+{
+
+  if (params.size() % 3 == 0) {
```

Comment:
Do you really need this check? Is this not checked much earlier already and then issuing a fatal error?

Reply:
It is indeed, this was an early attempt to make the method "more independent" from the only place that it can be really called. But I think that's a branch we can remove.

Reply:
yes, either remove this branch or throw an exception if this isn't fulfilled?

Reply:
I removed to not keep two checks.